DOC: update windows instructions and move conda/mamba

tupui opened the pull request:
Relates to #18806

This makes the instructions gotcha about building conda env on Windows more visible and move the note about mamba up too.

rgommers commented on doc/source/building/index.rst:
```diff
@@ -203,12 +203,26 @@ your system.
+       creation will not work due to an outdated Fortran compiler. If that
+       happens, remove the ``compilers`` entry from ``environment.yml`` and
+       try again. The Fortran compiler should be installed as described in
+       this section.
```
I can't check because the html didn't render, but this seems like the wrong place. The warning is specific to conda-on-Windows, but given the indentation level it may now always show up?

rgommers replied:
Also, it removes "on Windows", and I don't understand why. It should be Windows-specific.

tupui replied:
Should be fixed now. I had an indentation issue.

rgommers reviewed (changes requested):
The doc build failed for an unrelated reason, that would be good to fix first in a separate PR. Seeing the rendered html here is useful.

I'm wondering whether we should instead start shipping lock files like scikit-learn does. I think this tweak isn't an improvement, and a lot of folks are still going to miss it. Removing the possibility of running into this problem in the first place is probably better. WDYT?

tupui commented:
> The doc build failed for an unrelated reason, that would be good to fix first in a separate PR. Seeing the rendered html here is useful.

Builds locally now, let's see. I think it was just due to the indentation.

I also adjusted 2 other small things while I was there. Added a subheading in the build from source section and also swapped the pip/conda order in the last section to have the same order as the other section.

> I'm wondering whether we should instead start shipping lock files like scikit-learn does. I think this tweak isn't an improvement, and a lot of folks are still going to miss it. Removing the possibility of running into this problem in the first place is probably better. WDYT?

I am all for using lock files. I've always been doing that at previous companies for prod. I remember some discussions (and never followed up) saying `conda-lock` was still in the work? Is it good now to use? I've personally only experienced lock files through Poetry.

rgommers commented:
Thanks, I can see what changes now. The warning goes together specifically with this line:
```
mamba env create -f environment.yml
```
The warning was right under it, and now you've moved it far away. That seems wrong. I think you could either do nothing, or change `.. note::` to `.. warning::` - but leaving it in its current place.

tupui commented:
But the main idea was to move it so that we also see it in the case of building for simply using SciPy and not development.

rgommers commented:
> But the main idea was to move it so that we also see it in the case of building for simply using SciPy and not development.

At that point there is no need to use `environment.yml` at all, that is only for development purposes.

tupui commented:
Yes, in the "use" part we say that one can either use the environment file or manually install the minimal set of deps. And the list includes `compilers`. That's why I wanted to make the section more visible because otherwise a user is not going to read further (and another problem was that because the conda tab was not selected, you would really miss it if you don't continue and click the tab.)

rgommers reviewed (approved):
Okay, thanks for the explanation @tupui - I see how you can hit this outside of the contribution workflow now. Let's merge this and see if it helps. If it remains an issue for Windows users, I think we have to ship lock files (conda-lock should be fine indeed), or alternatively a separate `environment_win.yml`.
